File: PerfTester-Xcode/PerfTester-Xcode/Models/AssetLoadingTest.swift
//
//  AssetLoadingTest.swift
//  PerfTester-Xcode
//
//  Created by Patryk Średziński on 02.07.2018.
//  Copyright © 2018 private.perftesterxcode. All rights reserved.
//

import UIKit

class AssetLoadingTest: Test {
    
    private var assetName: String!
    
    override init() {
        super.init()
        title = "Asset Loading"
        desc = "Loading and Displaying 10 ImageViews"
        imageName = "Photo"
        parameters = [ 128, 256, 512, 1024, 2560, 5120, 10240 ]
    }
    
    override func prepare(param: Double) {
       assetName = String.init(format: "Asset%d", Int(param))
    }
    
    override func done(param: Double) {
        for view in self.detCtr.view.subviews {
            if view.tag == 3 {
                view.removeFromSuperview()
            }
        }
    }
    
    override func doJob(param: Double) {
        for i in 1...10 {
            let path = Bundle.main.path(forResource: self.assetName, ofType: "png")!
            let image = UIImage(contentsOfFile: path)
            //            let image = UIImage(named: self.assetName)
            let imageView = UIImageView(image: image)
            imageView.tag = 3
            imageView.frame = CGRect(x: i*10, y: i*20, width: 100, height: 100)
            self.detCtr.view.addSubview(imageView)
        }
        DispatchQueue.main.async {
            super.finishJob(param: param)
        }
    }
}

File: PerfTester-Xamarin/PerfTester-Xamarin/Models/AssetLoadingTest.cs
using System;
using CoreFoundation;
using PerfTesterXamarin.Screens.TestDetails;
using UIKit;

namespace PerfTesterXamarin.Models
{
    public class AssetLoadingTest : Test
    {
        public override string Title => "Asset Loading";
        public override string Desc => "Loading and Displaying 5 ImageViews";
        public override string ImageName => "Photo";
        public override double[] Parameters => new double[] { 128, 256, 512, 1024, 2560, 5120, 10240};

        string AssetName;

        public override void Prepare(double param)
        {
            AssetName = String.Format("Assets/Asset{0}.png", param);
        }

        public override void Done(double param)
        {
            this.DetCtr.TryGetTarget(out TestDetailsController detCtr);
            foreach (UIView view in detCtr.View.Subviews)
            {
                if (view.Tag == 3)
                {
                    view.RemoveFromSuperview();
                }
            }
        }

        public override void DoJob(double param)
        {

            this.DetCtr.TryGetTarget(out TestDetailsController detCtr);

            for (long i = 1; i <= 5; i++)
            {
                UIImage image = UIImage.FromFile(AssetName);

                UIImageView imageView = new UIImageView()
                {
                    Image = image,
                    Tag = 3,
                    Frame = new CoreGraphics.CGRect()
                    {
                        X = i * 20,
                        Y = i * 30,
                        Width = 100,
                        Height = 100
                    }
                };
                detCtr.View.AddSubview(imageView);
            }

            DispatchQueue.MainQueue.DispatchAsync(() =>
            {
                base.FinishJob(param);
            });

        }
    }
}